src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -242,6 +245,17 @@ export const TradeConfirm = () => {
     [sellTradeId, trade],
   )
 
+  const chainAdapterManager = getChainAdapterManager()
+  const failureAmounts = useMemo(
```

Comment:
Would it make sense to make this type `{}[]` instead of `{}[] | undefined` by returning an empty array when `undefined`? That way we could guarantee `failureAmounts` exists on a trade quote and map over it without optional chaining.

Reply:
seems like no, because the `ReceiveSummary` is also used in the `TradeInput` which never has this data availble. We should pass in an empty array but that would require `useMemo(90 => [], [])` to prevent re-renders (since its non-primitive)